enterprise/tasks.py

```diff
@@ -8,25 +8,60 @@
 from celery import shared_task
 from edx_django_utils.monitoring import set_code_owner_attribute
 
+from django.apps import apps
+from django.core import mail
 from django.db import IntegrityError
 
-from enterprise.models import EnterpriseCourseEnrollment, EnterpriseCustomerUser, EnterpriseEnrollmentSource
+from enterprise.utils import send_email_notification_message
 
 LOGGER = getLogger(__name__)
 
 
+@shared_task
+@set_code_owner_attribute
+def send_enterprise_email_notification(
+    enterprise_customer_uuid,
+    admin_enrollment,
+    email_items,
+):
+    """
+    Send enrollment email notifications to specified learners
+
+    Arguments:
+        * enterprise_customer_uuid (UUID)
+        * admin_enrollment=False : If True, this indicates admin based enrollment (e.g., bulk enrollment)
+        *
+        * email_items: list of dictionary objects with keys:
+        *   user (dict): a dict with either of the following forms:
+              - 1: { 'first_name': name, 'username': user_name, 'email': email } (similar to a User object)
+              - 2: { 'user_email' : user_email } (similar to a PendingEnterpriseCustomerUser object)
+        *   enrolled_in (dict): name and optionally other keys needed by templates
+        *   dashboard_url (str)
+    """
+    with mail.get_connection() as email_conn:
+        for item in email_items:
+            send_email_notification_message(
+                item['user'],
+                item['enrolled_in'],
+                item['dashboard_url'],
+                enterprise_customer_uuid,
+                email_connection=email_conn,
+                admin_enrollment=admin_enrollment,
+            )
+
+
 @shared_task
 @set_code_owner_attribute
 def create_enterprise_enrollment(course_id, enterprise_customer_user_id):
```

Comment:
Why were these changes needed? This is an unrelated shared task to the one you are modifying?

Reply:
oh these were because the direct use of the models in this task was causing circular ref errors since the model now calls the task

Reply:
ah just read your latest comment, yes will do that thanks